app/models/api_docs/service.rb

```diff
@@ -103,14 +141,15 @@ def specification
   end
 
   def api_product_production_public_base_url
-    return unless service&.proxy
-    service.proxy.endpoint
+    return if !owner.respond_to?(:proxy) || owner.proxy.blank?
+
+    owner.proxy.endpoint
   end
 
   private
 
   def service_belongs_to_account
-    return true if account.services.accessible.where(id: service_id).exists?
+    return true if account.services.accessible.where(id: service_id.presence || owner_id).exists?
```

Comment:
_Danger!_ If the `ApiDocs::Service` belongs to a `BackendApi`, this will cause a service trying to be found by the ID of the backend api.